File: Remote/Source/Remote Elements/REView/RemoteView.m
//
// RemoteView.m
// Remote
//
// Created by Jason Cardwell on 3/17/13.
// Copyright (c) 2013 Moondeer Studios. All rights reserved.
//
#import "RemoteElementView_Private.h"

static int ddLogLevel   = DefaultDDLogLevel;
static int msLogContext = (LOG_CONTEXT_REMOTE | LOG_CONTEXT_FILE | LOG_CONTEXT_CONSOLE);
#pragma unused(ddLogLevel, msLogContext)

@implementation RemoteView

- (CGSize)intrinsicContentSize { return [MainScreen bounds].size; }

/// Override sizing and moving of the remote view
////////////////////////////////////////////////////////////////////////////////

- (void)setResizable:(BOOL)resizable {}
- (void)setMoveable:(BOOL)moveable   {}
- (BOOL)isResizable { return NO; }
- (BOOL)isMoveable  { return NO; }


- (void)initializeIVARs {
  [self setContentCompressionResistancePriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisHorizontal];
  [self setContentCompressionResistancePriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisVertical];
  [self setContentHuggingPriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisHorizontal];
  [self setContentHuggingPriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisVertical];
  [super initializeIVARs];
}

@end

File: Remote/Source/Remote Elements/Views/RemoteView.swift
//
//  RemoteView.swift
//  Remote
//
//  Created by Jason Cardwell on 11/07/14.
//  Copyright (c) 2014 Moondeer Studios. All rights reserved.
//

import Foundation
import UIKit
import MoonKit

class RemoteView: RemoteElementView {

	/**
	intrinsicContentSize

	:returns: CGSize
	*/
	override func intrinsicContentSize() -> CGSize { return UIScreen.mainScreen().bounds.size }

	override var resizable: Bool { get { return false } set {} }
	override var moveable: Bool { get { return false } set {} }

	/** initializeIVARs */
	override func initializeIVARs() {
		setContentCompressionResistancePriority(1000.0, forAxis: .Horizontal)
		setContentCompressionResistancePriority(1000.0, forAxis: .Vertical)
		setContentHuggingPriority(1000.0, forAxis: .Horizontal)
		setContentHuggingPriority(1000.0, forAxis: .Vertical)
		super.initializeIVARs()
	}

}
